docs/template.md

```diff
@@ -3,4 +3,47 @@ title: "Template"
 weight: 40
 ---
 
->This document is under construction.
\ No newline at end of file
+## Template
```

Comment:
wdyt, we have this big diagram in https://github.com/asyncapi/generator/blob/master/docs/index.md and also simplified version for https://github.com/asyncapi/generator/blob/master/docs/asyncapi-document.md

I think it would make sense to have simplified `template` focused version here too. As AsyncAPI document and Template files are main building parts required by the generator?

[proposal]()https://mermaid-js.github.io/mermaid-live-editor/edit#pako:eNpVj70OwjAMhF8l8gwv0AGppcDCBGyEwRADEU0auakEKn133B-KyGTdfb6cG7iUhiCBG2O4q-1OeyUvPR7IhQIjqWXpIz3jaTCyZkOeGGPJ7aDkxx15Q6xW_mY9jdx6Cjip-Xyhsk6u6vPwzZShtvbMyK8xvEffCquXv2CwMqajMxgBGV31k9N-If_fnpqvbUEd3PvSEWbgiB1aI_c2nagh3smRhkRGg_zQoH0rXB2MJKyMlY6QXLGoaAZYx3IvzSCJXNMXyi3KUW6k2g9MUmuK
```
graph LR
    A[Template Context]
    B{Generator}
    D[Render Engine]
    F[Template] --> B
  subgraph Generator Library
    B --> | asyncapi | A
    B--> | params | A
    A --> D
    B --> | Template Files | D
  end
```

with shorter description like it is with `asyncapi document` doc

Reply:
totally makes sense👍